core/dbt/contracts/graph/manifest.py

```diff
@@ -437,15 +441,27 @@ def find(
 
         pkg_dct: Mapping[PackageName, List[Any]] = self.storage[search_name]
 
+        nodes = []
         if package is None:
             if not pkg_dct:
                 return None
             else:
-                return next(iter(pkg_dct.values()))
+                nodes = next(iter(pkg_dct.values()))
         elif package in pkg_dct:
-            return pkg_dct[package]
+            nodes = pkg_dct[package]
         else:
             return None
+        if not resource_types:
```

Comment:
an explicit `if resource_types is None` would be more precise here, since resource_types could also be `[]` which I think the more expected behaviour would be to return no node types as opposed to all of them.